src/Websocket.Client/WebsocketClient.cs

```diff
@@ -40,21 +40,48 @@ public class WebsocketClient : IWebsocketClient
 
         private readonly BlockingCollection<string> _messagesTextToSendQueue = new BlockingCollection<string>();
         private readonly BlockingCollection<byte[]> _messagesBinaryToSendQueue = new BlockingCollection<byte[]>();
-
-
+        
         /// <summary>
         /// A simple websocket client with built-in reconnection and error handling
         /// </summary>
         /// <param name="url">Target websocket url (wss://)</param>
         /// <param name="clientFactory">Optional factory for native ClientWebSocket, use it whenever you need some custom features (proxy, settings, etc)</param>
         public WebsocketClient(Uri url, Func<ClientWebSocket> clientFactory = null)
+            : this(url, GetClientFactory(clientFactory))
+        {
+
+        }
+
+        private static Func<Uri, CancellationToken, Task<WebSocket>> GetClientFactory(Func<ClientWebSocket> clientFactory)
+        {
+            if (clientFactory == null)
+                return null;
+
+            return (async (uri, token) => {
+                var client = clientFactory();
+                await client.ConnectAsync(uri, token).ConfigureAwait(false);
+                return client;
+            });
+        }
+
+        /// <summary>
+        /// A simple websocket client with built-in reconnection and error handling
+        /// </summary>
+        /// <param name="url">Target websocket url (wss://)</param>
+        /// <param name="connectionFactory">Optional factory for native ClientWebSocket, use it whenever you need some custom features (proxy, settings, etc)</param>
+        public WebsocketClient(Uri url, Func<Uri, CancellationToken, Task<WebSocket>> connectionFactory)
```

Comment:
It could be `internal`, you can then use attribute `[assembly: InternalsVisibleTo("Websocket.Client.Tests")]` to enable accessibility in unit tests. 
Nevertheless, that constructor could be useful to user, so we can leave it public. Still, it brings some implementation detail to the user and he has to know about calling `ConnectAsync()`. Could you please update XML comments above?

Reply:
I would like to have it public such that I can use it to do integration testing against TestHost.

I have updated the XML comments for this constructor

Reply:
I could also make a check in the StartClient method to validate if the Clients state is Open and if not throw an exception?

Reply:
Just leave it on XML comment only, an exception could be too much restrictive.